src/Servers/Kestrel/test/InMemory.FunctionalTests/ConnectionDispatcherTests.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO.Pipelines;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests.TestTransport;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+using Microsoft.AspNetCore.Testing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.InMemory.FunctionalTests
+{
+    public class ConnectionDispatcherTests : TestApplicationErrorLoggerLoggedTest
+    {
+        [Fact]
+        public async Task AppAbortViaConnectionContextIsLogged()
+        {
+            var testContext = new TestServiceContext(LoggerFactory);
+            var connectionDispatcher = new AbortingConnectionDispatcher();
```

Comment:
Nit: How is this test testing the ConnectionDispatcher? Seems more like you are mocking it and testing TransportConnection?